Source/QED/QuantumSyncEngineWrapper.cpp

```diff
@@ -14,4 +21,181 @@ QuantumSynchrotronEngine::build_optical_depth_functor ()
+
+    tie(is_ok, innards.ctrl.prob_tem_how_many, p_data) =
+        parse_raw_data<decltype(innards.ctrl.prob_tem_how_many)>(
+            p_data, p_last);
+    if(!is_ok) return false;
+
+    //___________________________
+
+    //Data
+    Vector<Real> tndt_coords(innards.ctrl.chi_part_tdndt_how_many);
+    Vector<Real> tndt_data(innards.ctrl.chi_part_tdndt_how_many);
+    Vector<Real> cum_tab_coords1(innards.ctrl.chi_part_tem_how_many);
+    Vector<Real> cum_tab_coords2(innards.ctrl.prob_tem_how_many);
+    Vector<Real> cum_tab_data(innards.ctrl.chi_part_tem_how_many*
+        innards.ctrl.prob_tem_how_many);
+
+    tie(is_ok, tndt_coords, p_data) =
+        parse_raw_data_vec<Real>(
+            p_data, tndt_coords.size(), p_last);
+    if(!is_ok) return false;
+    innards.KKfunc_coords.assign(tndt_coords.begin(), tndt_coords.end());
+
+    tie(is_ok, tndt_data, p_data) =
+        parse_raw_data_vec<Real>(
+            p_data, tndt_data.size(), p_last);
+    if(!is_ok) return false;
+    innards.KKfunc_data.assign(tndt_data.begin(), tndt_data.end());
+
+    tie(is_ok, cum_tab_coords1, p_data) =
+        parse_raw_data_vec<Real>(
+            p_data, cum_tab_coords1.size(), p_last);
+    if(!is_ok) return false;
+    innards.cum_distrib_coords_1.assign(
+        cum_tab_coords1.begin(), cum_tab_coords1.end());
+
+    tie(is_ok, cum_tab_coords2, p_data) =
+        parse_raw_data_vec<Real>(
+            p_data, cum_tab_coords2.size(), p_last);
+    if(!is_ok) return false;
+    innards.cum_distrib_coords_2.assign(
+        cum_tab_coords2.begin(), cum_tab_coords2.end());
+
+    tie(is_ok, cum_tab_data, p_data) =
+        parse_raw_data_vec<Real>(
+            p_data, cum_tab_data.size(), p_last);
+    if(!is_ok) return false;
+    innards.cum_distrib_data.assign(
+        cum_tab_data.begin(), cum_tab_data.end());
+
+    //___________________________
+    lookup_tables_initialized = true;
+
+    return true;
+}
+
+Vector<char> QuantumSynchrotronEngine::export_lookup_tables_data () const
+{
+    Vector<char> res{};
+
+    if(!lookup_tables_initialized)
+        return res;
+
+    auto add_data_to_buf = [&res](auto data){
+        res.insert(res.end(),
+        reinterpret_cast<const char*>(&data),
+        reinterpret_cast<const char*>(&data) +
+        sizeof(data));
+    };
+
+    add_data_to_buf(innards.ctrl.chi_part_min);
+    add_data_to_buf(innards.ctrl.chi_part_tdndt_min);
+    add_data_to_buf(innards.ctrl.chi_part_tdndt_max);
+    add_data_to_buf(innards.ctrl.chi_part_tdndt_how_many);
+    add_data_to_buf(innards.ctrl.chi_part_tem_min);
+    add_data_to_buf(innards.ctrl.chi_part_tem_max);
+    add_data_to_buf(innards.ctrl.chi_part_tem_how_many);
+    add_data_to_buf(innards.ctrl.prob_tem_how_many);
+
+    auto add_data_to_buf_vv = [&res](const auto* data, size_t how_many){
+        res.insert(res.end(),
+        reinterpret_cast<const char*>(data),
+        reinterpret_cast<const char*>(data) +
+        sizeof(*data)*how_many);
+    };
```

Comment:
Actually could this be in, e.g.,  `QedWrapperCommons.H`? Just a suggestion.

Reply:
yes. I've created a templated function:
```
    template <class T>
    void add_data_to_vector_char (
        const T* p_data, size_t how_many, amrex::Vector<char>& raw_data)
    {
        raw_data.insert(
            raw_data.end(),
            reinterpret_cast<const char*>(p_data),
            reinterpret_cast<const char*>(p_data) +
            sizeof(T)*how_many
        );
    }
```
and I've moved it to `QedTableParserHelperFunctions.H` .